src/earn/EarnEnterAmount.tsx

```diff
@@ -341,20 +342,17 @@ function EarnEnterAmount({ route }: Props) {
     }
   }
 
-  const onMaxAmountPress = async () => {
-    // eventually we may want to do something smarter here, like subtracting gas fees from the max amount if
-    // this is a gas-paying token. for now, we are just showing a warning to the user prompting them to lower the amount
-    // if there is not enough for gas
-    setTokenAmountInput(balanceInInputToken.toFormat({ decimalSeparator }))
+  const onSelectPercentageAmount = (percentage: number) => {
+    setTokenAmountInput(inputToken.balance.multipliedBy(percentage).toFormat({ decimalSeparator }))
```

Comment:
This breaks for withdraw flows. The balance for withdraw is different. We show the input token but the balance should be using the LP token balance (in units of input token)